pipeline-model-definition/src/main/resources/org/jenkinsci/plugins/pipeline/modeldefinition/generator/DirectiveGenerator/index.jelly

```diff
@@ -78,34 +78,10 @@
                 </f:dropdownList>
                 <j:set var="id" value="${h.generateId()}"/>
```

Comment:
Could probably get rid of the `generateId` trick now, by having the button look for a sibling textarea.

Reply:
Sibling in the sense- I can allocate a similar class name to the text area and remove all the mentions of the id variable?

Reply:
Would have to look up some example, since I do not normally work on JS at all, but IIRC the point is that now that the behavior function is passed an element it would be able to use that argument to perform operations without a (unique) ID—either directly on the element in the argument, or on some sibling in the DOM.

In other words: when using inline `<script>` with a Jelly facet that might be included multiple times on a page, like the config snippet for a build step in a freestyle project config screen, you have the problem that

```xml
<body>
  <!-- first occurrence -->
  <div>
    <button onclick="handle()"/>
    <textarea id="spot"/>
    <script>
      function handle() {
        document.getElementById('spot').text = 'handled';
      }
    </script>
  </div>
</body>
```

will not work as soon as you add a second `<div>` with the same content: clicking either button will update the same textarea, perhaps the wrong one. `generateId` used to be used as a workaround:

```xml
<body>
  <!-- first occurrence -->
  <div>
    <j:set var="id" value="${h.generateId()}"/>
    <button onclick="handle()"/>
    <textarea id="spot-${id}"/>
    <script>
      function handle() {
        document.getElementById('spot-${id}').text = 'handled';
      }
    </script>
  </div>
</body>
```

When switching to a proper adjunct, you do not need this trick since

```xml
<body>
  <!-- first occurrence -->
  <div>
    <button class="mybutton"/>
    <textarea/>
    <st:adjunct …/>
  </div>
</body>
```

```js
function handle(button) {
  button.getSiblingByType('textarea').text = 'handled';
}
```

will work regardless of how many copies there are. (All code above is pseudocode, actual method names etc. are going to be a bit different.)

Of course in many cases the Jelly fragment could never be included more than once in a page because of what it is for, so whether a unique id was even needed here to begin with, I am unsure; it may have been included “just in case”.

Reply:
Ohhhh makes sense. Thank you, will treat it as nit for future contributions in Jenkins.